os/arch/arm/include/armv7-r/irq.h

```diff
@@ -77,17 +77,20 @@
  * TODO: This definition should be moved to platform specific header
  *
  */
-#define MPU_REG_ENTIRE_MAP    (0)
-#define MPU_REG_USER_RAM    (1)
-#define MPU_REG_USER_CONFIG1  (2)
-#define MPU_REG_USER_CONFIG0  (3)
-#define MPU_REG_TASK_STACK    (4)
-#define MPU_REG_KERN_FLASH    (5)
-#define MPU_REG_KERN_DATA   (6)
-#define MPU_REG_KERN_VEC    (7)
-
-/* Total number of regions for user, including the user task region */
-#define MPU_TOTAL_USER_REG          (MPU_REG_KERN_FLASH - MPU_REG_USER_CONFIG1)
+#define MPU_REG0       (0)
+#define MPU_REG1       (1)
+#define MPU_REG2       (2)
+#define MPU_REG3       (3)
+#define MPU_REG4       (4)
+#define MPU_REG5       (5)
+#define MPU_REG6       (6)
+#define MPU_REG7       (7)
+#if (CONFIG_ARMV7M_MPU_NREGIONS == 12)
```

Comment:
Why did you use the ARMV7M option here? V7M? V7R?

Reply:
please check the arch/arm/Kconfig. It is a common definition for MPU. Doesn't depend on any cpu architecture.  Then name is carried away from NuttX, we can fix it to use a common name ARM_MPU. But not in this patchset. That can be pushed as a separate patch

Reply:
@girishks2000 
For example, artik053 only uses Cortex-R4, not Cortex-M0. So I am going to remove ARMV7M from defconfig of all artik053. But you do not have any problems?

Reply:
If artik53 has to be built in protect mode, then we need it. It is not dependent on M or R.

Reply:
Ok in artik53, how do you define number of regions. can you please tell me how you will configure the number of regions?

Reply:
@girishks2000 
Since artik053 does not use cortex-m0, it can be changed as follows.
```diff
--- a/build/configs/artik053/nettest/defconfig
+++ b/build/configs/artik053/nettest/defconfig
@@ -144,8 +144,7 @@ CONFIG_ARCH_CORTEXR4=y
 CONFIG_ARCH_FAMILY="armv7-r"
 CONFIG_ARCH_CHIP="s5j"
 # CONFIG_ARCH_HAVE_FPU is not set
-CONFIG_ARMV7M_MPU=y
-CONFIG_ARMV7M_MPU_NREGIONS=12
+# CONFIG_ARMV7M_MPU is not set
```

Reply:
So, this config is needed for artik053 and sidk_s5jt200. We can't disable this. Anyway as Girish said, we should rename it at different patch, not in this patch.